app/code/Magento/QuoteGraphQl/Model/Cart/PlaceOrder.php

```diff
@@ -58,6 +58,26 @@ public function execute(Quote $cart, string $maskedCartId, int $userId): int
         $cartId = (int)$cart->getId();
         $paymentMethod = $this->paymentManagement->get($cartId);
 
+        // Get a list of available payment methods for the cart
+        $availablePaymentMethods = $this->paymentManagement->getList($cartId);
+        $payment = $cart->getPayment();
+        $paymentMethodCode = $payment?->getMethod();
+        $isPaymentMethodAvailable = false;
+
+        // Check if the selected payment method is in the available methods list
+        if ($paymentMethodCode && $availablePaymentMethods) {
+            foreach ($availablePaymentMethods as $availableMethod) {
+                if ($availableMethod->getCode() === $paymentMethodCode) {
+                    $isPaymentMethodAvailable = true;
+                    break;
+                }
+            }
+        }
```

Comment:
Suggesting the below changes:
```suggestion
        // Check if the selected payment method is in the available methods list
        if ($paymentMethodCode && $availablePaymentMethods) {
            $availableCodes = array_map(fn($method) => $method->getCode(), $availablePaymentMethods);
            $isPaymentMethodAvailable = in_array($paymentMethodCode, $availableCodes);
        } else {
            $isPaymentMethodAvailable = false;
        }
```